internal/inventory/awsfetcher/fetcher_iam_role.go

```diff
@@ -68,20 +68,16 @@ func (i *iamRoleFetcher) Fetch(ctx context.Context, assetChannel chan<- inventor
 
 		assetChannel <- inventory.NewAssetEvent(
 			inventory.AssetClassificationAwsIamRole,
-			[]string{pointers.Deref(role.Arn), pointers.Deref(role.RoleId)},
```

Comment:
maybe `role.RoleId` needs to go in `WithRelatedAssetIds` ? same for all other instances where we used to pass a an array of strings as first argument and now just pass the resource ARN without the other fields

Reply:
That's a good point, I'll update